src/support/hash.h

```diff
@@ -45,6 +45,11 @@ inline uint64_t rehash(uint64_t x, uint64_t y) {
   return rehash(ret, HashType(y >> 32));
 }
 
+template<typename T> inline void hash_combine(std::size_t& s, const T& v) {
+  // see boost/container_hash/hash.hpp
+  s ^= std::hash<T>{}(v) + 0x9e3779b9 + (s << 6) + (s >> 2);
```

Comment:
It seems `std::hash` is [not the best hash function](https://stackoverflow.com/a/50978188). I guess using xxHash will be better in term of speed and quality

Reply:
Read similar things, but didn't know how important it is for this PR. Perhaps as a follow-up?

Reply:
Hmm, I don't know. But will be good keep this in mind